facefusion/workflows/core.py

```diff
@@ -20,3 +27,62 @@ def clear() -> ErrorCode:
 	clear_temp_directory(state_manager.get_temp_path(), state_manager.get_item('output_path'))
 	logger.debug(translator.get('clearing_temp'), __name__)
 	return 0
+
+
+def conditional_get_source_audio_frame(frame_number : int) -> AudioFrame:
+	if state_manager.get_item('workflow') == 'image-to-image':
+		return create_empty_audio_frame()
+	source_audio_path = get_first(filter_audio_paths(state_manager.get_item('source_paths')))
+	output_video_fps = state_manager.get_item('output_video_fps')
+
+	if state_manager.get_item('workflow') == 'image-to-video':
+		output_video_fps = restrict_video_fps(state_manager.get_item('target_path'), state_manager.get_item('output_video_fps'))
+	return get_audio_frame(source_audio_path, output_video_fps, frame_number)
+
+
+def conditional_get_source_voice_frame(frame_number: int) -> AudioFrame:
+	if state_manager.get_item('workflow') == 'image-to-image':
+		return create_empty_audio_frame()
+	source_audio_path = get_first(filter_audio_paths(state_manager.get_item('source_paths')))
+	output_video_fps = state_manager.get_item('output_video_fps')
+
+	if state_manager.get_item('workflow') == 'image-to-video':
+		output_video_fps = restrict_video_fps(state_manager.get_item('target_path'), state_manager.get_item('output_video_fps'))
+	return get_voice_frame(source_audio_path, output_video_fps, frame_number)
+
+
+def conditional_get_reference_vision_frame() -> VisionFrame:
+	if state_manager.get_item('workflow') == 'image-to-video':
+		return read_static_video_frame(state_manager.get_item('target_path'), state_manager.get_item('reference_frame_number'))
+	return read_static_image(state_manager.get_item('target_path'))
+
+
+def process_temp_frame(temp_frame_path : str, frame_number : int) -> bool:
+	reference_vision_frame = conditional_get_reference_vision_frame()
+	source_vision_frames = read_static_images(state_manager.get_item('source_paths'))
+	target_vision_frame = read_static_image(temp_frame_path, 'rgba')
+	temp_vision_frame = target_vision_frame.copy()
+	temp_vision_mask = extract_vision_mask(temp_vision_frame)
+
+	source_audio_frame = conditional_get_source_audio_frame(frame_number)
+	source_voice_frame = conditional_get_source_voice_frame(frame_number)
+
+	if not numpy.any(source_audio_frame):
```

Comment:
remove this both empty check block, this goes to the `conditional_get_source_audio_frame` and `conditional_get_source_voice_frame`